src/tools/r2rdump/R2RMethod.cs

```diff
@@ -152,44 +153,42 @@ public enum GenericElementTypes
         /// <summary>
         /// Extracts the method signature from the metadata by rid
         /// </summary>
-        public R2RMethod(byte[] image, MetadataReader mdReader, uint rid)
+        public R2RMethod(byte[] image, MetadataReader mdReader, uint rid, int entryPointId, GenericElementTypes[] instanceArgs, uint[] tok)
         {
+            Token = _mdtMethodDef | rid;
+            EntryPointRuntimeFunctionId = entryPointId;
+
             _mdReader = mdReader;
             NativeCode = new List<RuntimeFunction>();
 
             // get the method signature from the MethodDefhandle
             MethodDefinitionHandle methodDefHandle = MetadataTokens.MethodDefinitionHandle((int)rid);
-            MethodDefinition methodDef = mdReader.GetMethodDefinition(methodDefHandle);
-            Name = mdReader.GetString(methodDef.Name);
-            BlobReader signatureReader = mdReader.GetBlobReader(methodDef.Signature);
+            _methodDef = mdReader.GetMethodDefinition(methodDefHandle);
+            Name = mdReader.GetString(_methodDef.Name);
+            BlobReader signatureReader = mdReader.GetBlobReader(_methodDef.Signature);
 
-            DeclaringType = new Stack<string>();
-            TypeDefinitionHandle declaringTypeHandle = methodDef.GetDeclaringType();
+            TypeDefinitionHandle declaringTypeHandle = _methodDef.GetDeclaringType();
             TypeDefinition declaringTypeDef;
             while (!declaringTypeHandle.IsNil)
             {
                 declaringTypeDef = mdReader.GetTypeDefinition(declaringTypeHandle);
-                DeclaringType.Push(mdReader.GetString(declaringTypeDef.Name));
+                DeclaringType = mdReader.GetString(declaringTypeDef.Name) + "." + DeclaringType;
                 declaringTypeHandle = declaringTypeDef.GetDeclaringType();
             }
             
             NamespaceDefinitionHandle namespaceHandle = declaringTypeDef.NamespaceDefinition;
             while (!namespaceHandle.IsNil)
             {
                 NamespaceDefinition namespaceDef = mdReader.GetNamespaceDefinition(namespaceHandle);
-                DeclaringType.Push(mdReader.GetString(namespaceDef.Name));
+                DeclaringType = mdReader.GetString(namespaceDef.Name) + "." + DeclaringType;
                 namespaceHandle = namespaceDef.Parent;
             }
 
             SignatureHeader signatureHeader = signatureReader.ReadSignatureHeader();
             IsGeneric = signatureHeader.IsGeneric;
-            var genericParams = methodDef.GetGenericParameters();
-            _genericParamInstance = new Dictionary<string, GenericElementTypes>();
-            foreach (GenericParameterHandle genericParam in genericParams)
-            {
-                _genericParamInstance[mdReader.GetString(mdReader.GetGenericParameter(genericParam).Name)] = 0;
-            }
-
+            var genericParams = _methodDef.GetGenericParameters();
```

Comment:
nit: Could you write the type GenericParameterHandleCollection instead of var here?